app/assets/javascripts/channels/groups.js

```diff
@@ -0,0 +1,131 @@
+function initGroupCable() {
+  var messages = $('#group-messages');
+  if(messages.length <= 0) return;
+
+  var groupId = messages.data('group-id');
+  var userId = messages.data('user-id');
+  var template = JST['templates/message'];
+
+  var page = 2;
+  var totalPages = 37;
+  var loadingMessages = false;
+
+  var toBottom = function() {
+    messages.scrollTop(messages.prop("scrollHeight"));
+  };
+
+  var prependMessage = function(message) {
+    messages.prepend(template({message: message, current_user: userId}));
+  };
+
+  var loadMessages = function() {
+    $.getJSON('/grupper/' + groupId + '/meddelanden/')
+      .then(function(resp) {
+        for (var message of resp.messages) prependMessage(message);
+        totalPages = resp.meta.total_pages;
+        toBottom();
+      });
+  }();
+
+  var appendMessage = function(message) {
+    messages.append(template({message: message, current_user: userId}));
+    toBottom();
+  };
+
+  var removeMessage = function(messageId) {
+    $('.message#' + messageId).remove();
+  };
+
+  var updateMessage = function(message) {
+    // If the message is loaded, update it!
+    var msg = $('.message#' + message.id);
+    if(!msg.length) return;
+
+    msg.find('.msg-content').html(message.content);
+    msg.find('.updated').html(message.updated_at);
+  };
+
+  var moreMessages = function() {
+    // Keep a reference to the current top element
+    var topMsg = $('.message:first');
+
+    $.getJSON('/grupper/' + groupId + '/meddelanden?page=' + page)
+      .then(function(resp) {
+        for (var message of resp.messages) {
+          prependMessage(message);
+        }
+
+        // Scroll to the old top message
+        messages.scrollTop(topMsg.position().top);
+
+        page++;
+        loadingMessages = false;
+      });
+  };
+
+  App.groups = App.cable.subscriptions.create({
+    channel: "GroupsChannel",
+    group_id: groupId },
+  { connected: function() {},
+    disconnected: function() {},
+    received: function(data) {
+      if (data.action == 'create') {
+        appendMessage(data.message.message);
+      } else if (data.action == 'destroy') {
+        removeMessage(data.message_id);
```

Comment:
Variables that come from the json data created in rails. Not going to edit that for some JS guidelines